controllers/provider-packet/pkg/apis/config/helper/helper.go

```diff
@@ -18,17 +18,28 @@ import (
 	"fmt"
 
 	"github.com/gardener/gardener-extensions/controllers/provider-packet/pkg/apis/config"
+	api "github.com/gardener/gardener-extensions/controllers/provider-packet/pkg/apis/packet"
 )
 
 // FindImage takes a list of machine images, and the desired image name and version. It tries
 // to find the image with the given name and version. If it cannot be found then an error
 // is returned.
-func FindImage(machineImages []config.MachineImage, imageName, version string) (string, error) {
-	for _, machineImage := range machineImages {
-		if machineImage.Name == imageName && machineImage.Version == version {
+func FindImage(profileImages []api.MachineImages, configImages []config.MachineImage, imageName, imageVersion string) (string, error) {
```

Comment:
```suggestion
func FindImage(profileImages []api.MachineImages, imageName, imageVersion string) (string, error) {
```

As discussed, let's only fetch the images from the `CloudProfileConfig`. Also, move this file to `pkg/apis/packet/helper`.

Reply:
not yet resolved